### ISSUE TITLE
hcloud_server fails due to rate limit of requests per hour

### ISSUE THREAD

**Issue by dongsupark** [hetznercloud/terraform-provider-hcloud#30]
While trying to create servers by running `terraform apply`, I often get the following error:

```
hcloud_server.workers.1: limit of 3600 requests per hour reached (rate_limit_exceeded) 
```

It's exactly the same error described in the [hetzner document](https://docs.hetzner.cloud/#header-rate-limiting-1).

The thing is, in practice "3600 requests per hour" is a very tight limitation. I defined only a 6-nodes cluster, (3 controllers, 3 workers). One single `apply` action takes around 7 - 8 minutes, and `destroy` takes 2 - 3 minutes. When I repeat a round of tests (apply && destroy) only twice, then it reaches already nearly up to the limit of 3600 req/h. Thus when I run `apply` again, then it fails with the error above. It's very easy to hit the limit inside one hour.

How can I raise the limit to avoid the limit error?

**Comment by thetechnick** [comment]
Hi @dongsupark,
we have just released a new version of our `hcloud-go` lib, this update includes updated rate limit handling, so these error should not occur anymore.
The update is included in https://github.com/hetznercloud/terraform-provider-hcloud/releases/tag/v1.1.0

**Comment by dongsupark** [comment]
@thetechnick Thanks for your quick reply.
The fix looks like this commit https://github.com/hetznercloud/hcloud-go/commit/b26c750a5ac142460e2666002c0563604baf457d.
Indeed it seems to have been fixed in terraform-provider-hcloud v1.1.0. I have been heavily testing (destroy && create) with the new version, and it does not fail any more.
Thanks a lot!
